src/transport_udp.cpp

```diff
@@ -242,361 +520,480 @@ UDPTransport::fd_writer()
+    if (data_ctx_it == conn_it->second->data_contexts.end()) {
+        // Invalid data context id
+        return TransportError::InvalidDataContextId;
+    }
 
-        Addr r;
-        r.key = ra_key;
-        r.addr_len = remoteAddrLen;
-        memcpy(&(r.addr), &remoteAddr, remoteAddrLen);
+    ConnData cd;
+    cd.data = bytes;
+    cd.conn_id = conn_id;
+    cd.data_ctx_id = data_ctx_id;
+    cd.priority = priority;
 
-        ++last_context_id;
-        ++last_stream_id;
+    data_ctx_it->second.tx_data->push(cd, ttl_ms, priority);
 
-        remote_contexts[last_context_id] = r;
-        remote_addrs[ra_key] = {
-          last_context_id,
-            last_stream_id,
-        };
+    return TransportError::None;
+}
 
-        cd.contextId = last_context_id;
-        cd.streamId = last_stream_id;
+std::optional<std::vector<uint8_t>> UDPTransport::dequeue(const TransportConnId &conn_id,
+                                                          const DataContextId &data_ctx_id) {
 
-        // Create dequeue
-        dequeue_data_map[last_context_id][last_stream_id].set_limit(1000);
+    const auto conn_it = conn_contexts.find(conn_id);
+    if (conn_it == conn_contexts.end()) {
+        logger->warning << "dequeue: invalid conn_id: " << conn_id
+                        << std::flush;
+        // Invalid context id
+        return std::nullopt;
+    }
 
-        cd.contextId = last_context_id;
+    const auto data_ctx_it = conn_it->second->data_contexts.find(data_ctx_id);
+    if (data_ctx_it == conn_it->second->data_contexts.end()) {
+        logger->error << "dequeue: invalid stream for conn_id: " << conn_id
+                      << " data_ctx_id: " << data_ctx_id << std::flush;
 
-        // Notify caller that there is a new connection
-        delegate.on_new_connection(last_context_id, std::move(remote));
+        return std::nullopt;
+    }
 
-      } else {
-        // Client mode doesn't support creating connections based on received
-        // packets
-        continue;
-      }
-    } else {
-      auto sctx = remote_addrs[ra_key];
-      cd.contextId = sctx.tcid;
-      cd.streamId = sctx.sid;
+    if (auto cd = data_ctx_it->second.rx_data.pop()) {
+        return cd.value().data;
     }
 
-    // Add data to caller queue for processing
-    auto& dq = dequeue_data_map[cd.contextId][cd.streamId];
+    return std::nullopt;
+}
 
-    // TODO: Notify caller that packets are being dropped on queue full
-    dq.push(cd);
+TransportConnId UDPTransport::connect_client() {
+    std::stringstream s_log;
 
-    if (dq.size() < 2) {
-      // Notify the caller that there is data to process
-      delegate.on_recv_notify(cd.contextId, cd.streamId);
+    fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+    if (fd == -1) {
+        throw std::runtime_error("socket() failed");
     }
-  }
 
-  logger->Log("Done transport reader thread");
-}
+    size_t snd_rcv_max = 64000;   // TODO: Add config for value
```

Comment:
I saw a couple of places where 64000 was explicitly referenced. I'd suggest making this `constexpr size_t Max_Packet_Size = 65535` or something until we sort out the dynamic configuration.

Reply:
Good suggestion, updated.